spec/lib/email/receiver_spec.rb

```diff
@@ -756,6 +756,16 @@ def create_post_reply_key(value)
       MD
     end
 
+    it "tries not to repeat duplicate attachments" do
+      SiteSetting.authorized_extensions = "jpg"
+
+      expect { process(:logo_1) }.to change { UploadReference.count }.by(1)
+      expect(topic.posts.last.raw).to match %r{upload://}
+
+      expect { process(:logo_2) }.not_to change { UploadReference.count }
```

Comment:
`logo_2` is a reply to `logo_1` and also include the same logo at attachment. There should not be a new `UploadReference` record created since it's a duplicate attachment.